### PR TITLE
Fix deprovisioning

### LINKED ISSUE #181: Assure cleanup of PVs
Assure cleanup of PVs after deleting of kyma cluster.

Right now when the kyma cluster is deleted all the PVs are left. (9 disks, about 250GB for each cluster so about 10$ per month). 
It should be also assured in periodic cleanup of leftovers described here https://github.com/kyma-project/test-infra/issues/177

### LINKED ISSUE #226: Change name of clusters for kyma-gke-integration jobs
For now, these clusters have a calculated (fixed) name - always the same for given Pull Request.
This causes name clashes when re-running job (cluster already exists).
Re-running job happens often, sometimes because of manual action, sometimes Prow itself does it.
We could try to ensure clusters are properly deleted, but it's hard in case of errors or when Prow job is just killed.
The easiest solution is to avoid name clashes by naming clusters randomly.

AC:
- change name of clusters to the one that won't conflict on re-tries (and with other jobs). Proposal: [PR_NUMBER]-UUID

### PR DESCRIPTION
**Pull request by Tomasz-Smelcerz-SAP**
**Description**

Changes proposed in this pull request:

- Clean up orphaned disks from gcloud at the end of gke-integration job
- Change cluster names to be unique: retires of the job do not fail because of a name conflict

**Related issue(s)**
Fixes #181
Fixes #226

### CODE REVIEW

File: prow/scripts/cluster-integration/delete-disks.sh [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+# DISKS=$(kubectl get pvc --all-namespaces -o jsonpath="{.items[*].spec.volumeName}" | xargs -n1 echo)
+# This allows to get a gcloud disk name suffix (not entire name), which is enough to find the proper object and delete it.
+#
+#Expected vars:
+# - DISKS: list of gcloud disk names (name can be a name suffix)
+#
+#Permissions: In order to run this script you need to use a service account with "Compute Admin" role
+
+set +e
+echo "Removing remaining PVC disks"
```
Comment (suleymanakbas91) [inline prow/scripts/cluster-integration/delete-disks.sh]: We can use shout here to make this part more visible in the logs.

File: prow/scripts/cluster-integration/pr-gke-integration.sh [hunk 2]
```diff
@@ -40,10 +40,10 @@ if [ "${discoverUnsetVar}" = true ] ; then
     exit 1
 fi
 
-trap cleanup EXIT
+#trap cleanup EXIT
 
 #!Put cleanup code in this function!
-cleanup() {
+cleanup11() {
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/pr-gke-integration.sh]: probably wrong function name

File: prow/scripts/cluster-integration/pr-gke-integration.sh [hunk 3]
```diff
@@ -40,10 +40,10 @@ if [ "${discoverUnsetVar}" = true ] ; then
     exit 1
 fi
 
-trap cleanup EXIT
+#trap cleanup EXIT
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/pr-gke-integration.sh]: why is it commented?

File: prow/scripts/cluster-integration/pr-gke-integration.sh [hunk 4]
```diff
@@ -56,11 +56,18 @@ cleanup() {
...
     if [ -n "${CLEANUP_CLUSTER}" ]; then
+        #save disk names while the cluster still exists to remove them later
+        DISKS=$(kubectl get pvc --all-namespaces -o jsonpath="{.items[*].spec.volumeName}" | xargs -n1 echo)
+        export DISKS
         shout "Deprovision cluster: \"${CLUSTER_NAME}\""
         date
         "${TEST_INFRA_SOURCES_DIR}"/prow/scripts/cluster-integration/deprovision-gke-cluster.sh
         TMP_STATUS=$?
         if [[ ${TMP_STATUS} -ne 0 ]]; then EXIT_STATUS=${TMP_STATUS}; fi
+        sleep 5
```
Comment (aszecowka) [inline prow/scripts/cluster-integration/pr-gke-integration.sh]: why sleep is needed?

Comment (Tomasz-Smelcerz-SAP) [reply prow/scripts/cluster-integration/pr-gke-integration.sh]: Removed

File: prow/scripts/cluster-integration/pr-gke-integration.sh (hunk 2 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/scripts/cluster-integration/pr-gke-integration.sh]: Fixed

File: prow/scripts/cluster-integration/pr-gke-integration.sh (hunk 3 above)
Comment (Tomasz-Smelcerz-SAP) [reply prow/scripts/cluster-integration/pr-gke-integration.sh]: Fixed

**Comment by aszecowka** [comment]
```
TODO: DEBUG: DISKS: pvc-5820bff7-f3d5-11e8-b033-42010a9c00e7
pvc-58200e35-f3d5-11e8-b033-42010a9c00e7
pvc-595336db-f3d5-11e8-b033-42010a9c00e7
pvc-e04944d9-f3d5-11e8-b033-42010a9c00e7
pvc-f828691e-f3d4-11e8-b033-42010a9c00e7
pvc-046192f4-f3d5-11e8-b033-42010a9c00e7
pvc-0f20c884-f3d5-11e8-b033-42010a9c00e7
pvc-59023194-f3d5-11e8-b033-42010a9c00e7
pvc-5bea3daa-f3d5-11e8-b033-42010a9c00e7
```
Please fix those logs

**Pull request merged**